ipam/ring/ring.go

```diff
@@ -282,29 +300,57 @@ func (es entries) merge(other entries, ourPeer mesh.PeerName) (result entries, u
 		case mine.Token > theirs.Token:
 			// insert, checking that a range owned by us hasn't been split
 			if previousOwner != nil && *previousOwner == ourPeer && theirs.Peer != ourPeer {
-				err = errEntryInMyRange(theirs)
-				return
+				// check we have no allocations in the range that got split
+				if checkEntryHasAllocations(es.entry(i-1), mine) {
```

Comment:
This doesn't seem quite right - suppose the tokens are A ... B ... C, and `theirs.Token` is B.
We want to check if there are allocations in the range B..C, but this code checks A..C.

Reply:
This is the case of range that got split. Its more like `mine` sees A...C and and `theirs.Token` is B with A<B< C. The check is at the point where `mine.Token` is C and `theirs.Token` is B and `checkEntryHasAllocations(es.entry(i-1), mine)` is checking over range A-C.

Reply:
That's what I meant to say - we don't mind if there are allocations in `A..B`, since that's staying with the same ownership; we want to check in `B..C` only since that's the bit that's changing.

Reply:
got it. Modified the check to only to range `B..C`